applications/base/skills.py

```
@@ -534,21 +1008,269 @@ class industry_test_implementation(BaseTool):
+    perform_ocr: bool = Field(default=False, description="True if you want the 4o model to read/analyze the text in the camera feed.")
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+class analyze_camera_frame_implementation(BaseTool):
+    """Analyze the camera frame and return the result based on the prompt."""
+    name: ClassVar[str] = "analyze_camera_frame"
+    description: ClassVar[str] = "The tool to see what is around the robot from camera"
+    args_schema: ClassVar[Type[BaseModel]] = analyze_camera_frame_definition
+    return_direct: ClassVar[bool] = False
+    verbose: ClassVar[bool] = False
+
+    def _run(self, prompt: str = None, camera_name: str = "camera", perform_ocr: bool = False):
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)